lib/matplotlib/artist.py

```diff
@@ -93,6 +100,56 @@ class Artist:
 
     zorder = 0
 
+    def __init_subclass__(cls):
+        # Inject custom set() methods into the subclass with signature and
+        # docstring based on the subclasses' properties.
+
+        if not hasattr(cls.set, '_autogenerated_signature'):
+            # Don't overwrite cls.set if the subclass or one of its parents
+            # has defined a set method set itself.
+            # If there was no explicit definition, cls.set is inherited from
+            # the hierarchy of auto-generated set methods, which hold the
+            # flag _autogenerated_signature.
+            return
+
+        def set(self, **kwargs):
+            Artist.set(self, **kwargs)
+
+        cls.set = set
+        cls.set.__qualname__ = f"{cls.__qualname__}.set"
+        cls._update_set_signature_and_docstring()
+
+    _PROPERTIES_EXCLUDED_FROM_SET = [
+        'navigate_mode',  # not a user-facing function
+        'figure',         # changing the figure is such a profound operation
+                          # that we don't want this in set()
+        '3d_properties',  # cannot be used as a keyword do to leading digit
+    ]
+
+    @classmethod
+    def _update_set_signature_and_docstring(cls):
+        """
+        Update the signature of the set function to list all properties
+        as keyword arguments.
+
+        Property aliases are not listed in the signature for brevity, but
+        are still accepted as keyword arguments.
+        """
+        cls.set.__signature__ = Signature(
+            [Parameter("self", Parameter.POSITIONAL_OR_KEYWORD),
+             *[Parameter(prop, Parameter.KEYWORD_ONLY, default=UNSET)
+               for prop in ArtistInspector(cls).get_setters()
+               if prop not in Artist._PROPERTIES_EXCLUDED_FROM_SET]])
+        cls.set._autogenerated_signature = True
+
+        cls.set.__doc__ = "\n".join([
+            "Set multiple properties at once.",
+            "",
+            "Supported properties are:",
+            "",
+            f"%({cls.__name__}:kwdoc)s"])
+        docstring.dedent_interpd(cls.set)
```

Comment:
I guess I would write this without going through `docstring`:
```python
cls.set.__doc__ = (
    "Set multiple properties at once.\n\n"
    "Supported properties are\n\n"
    + kwdoc(cls))
```

Reply:
yup.